runtime/ops/mod.rs

```diff
@@ -61,6 +62,26 @@ where
   rt.register_op(name, metrics_op(name, json_op_sync(op_fn)));
 }
 
+pub fn reg_buffer_async<F, R, RV>(
+  rt: &mut JsRuntime,
+  name: &'static str,
+  op_fn: F,
+) where
+  F: Fn(Rc<RefCell<OpState>>, u32, BufVec) -> R + 'static,
+  R: Future<Output = Result<RV, AnyError>> + 'static,
+  RV: ValueOrVector,
+{
+  rt.register_op(name, metrics_op(name, buffer_op_async(op_fn)));
+}
+
+pub fn reg_buffer_sync<F, R>(rt: &mut JsRuntime, name: &'static str, op_fn: F)
+where
+  F: Fn(&mut OpState, u32, &mut [ZeroCopyBuf]) -> Result<R, AnyError> + 'static,
+  R: ValueOrVector,
+{
+  rt.register_op(name, metrics_op(name, buffer_op_sync(op_fn)));
+}
```

Comment:
I like how this is mirrors the structure of the JSON ops register function - so I think easier to understand.